### PR TITLE
fix(httpclient): handle invalid chunk size in chunked transfer decoding

### LINKED ISSUE #129: httpclient: ValueError in _aiter_chunked when upstream injects raw HTTP error mid-stream
## Bug

`_aiter_chunked` crashes with `ValueError` when an upstream reverse proxy injects a raw HTTP error response into an active chunked transfer stream.

### Reproduction

When the upstream (e.g. nginx or a reverse proxy) encounters an error mid-stream during a chunked response, it may inject a raw HTTP status line like `HTTP/1.1 502 Bad Gateway` directly into the chunked data. The `_aiter_chunked` method reads this line expecting a hex chunk size and crashes:

```
ValueError: invalid literal for int() with base 16: 'HTTP/1.1 502 Bad Gateway'
```

### Traceback (from llm-rosetta-gateway production)

```
File "llm_rosetta/_vendor/httpclient.py", line 832, in aiter_lines
File "llm_rosetta/_vendor/httpclient.py", line 817, in _aiter_chunked
    chunk_size = int(size_str, 16)
ValueError: invalid literal for int() with base 16: 'HTTP/1.1 502 Bad Gateway'
```

### Root cause

`_aiter_chunked` at line 817 does:

```python
chunk_size = int(size_str, 16)
```

No validation is done to check whether `size_str` is actually a valid hex number. When the upstream breaks protocol, this produces a confusing `ValueError` instead of a clear connection error.

### Expected behavior

When the chunk size line is not valid hex, raise a descriptive error (e.g. `ConnectionError` or `HTTPError`) that includes the raw line content, so callers can understand that the upstream broke the chunked encoding protocol.

### Environment

- httpclient version: 0.4.5
- Observed in: llm-rosetta-gateway production (upstream was argo-proxy returning 502)

### PR DESCRIPTION
**Pull request by Oaklight**
## Summary

- Catch `ValueError` in `_aiter_chunked` when the chunk size line is not valid hex, and raise `HttpConnectionError` with the raw line content
- Handles the case where an upstream reverse proxy injects a raw HTTP error response (e.g. `HTTP/1.1 502 Bad Gateway`) into an active chunked stream
- Add 4 edge-case tests for chunked transfer decoding (valid stream, mid-stream HTTP error, garbage chunk size, error after valid chunks)

Closes #129

## Test plan

- [x] `TestChunkedDecoding::test_valid_chunked_stream` — normal chunked stream still works
- [x] `TestChunkedDecoding::test_http_error_injected_mid_stream` — HTTP error line raises `HttpConnectionError`
- [x] `TestChunkedDecoding::test_garbage_chunk_size` — non-hex line raises `HttpConnectionError`
- [x] `TestChunkedDecoding::test_error_after_valid_chunks` — error after valid data raises correctly, prior chunks still yielded
- [x] Full httpclient test suite (176 tests) passes

### CODE REVIEW

File: httpclient/httpclient.py [hunk 1]
```diff
@@ -814,7 +814,13 @@ async def _aiter_chunked(self) -> AsyncIterator[bytes]:
...
                 break
-            chunk_size = int(size_str, 16)
+            try:
+                chunk_size = int(size_str, 16)
+            except ValueError:
+                raise HttpConnectionError(
+                    f"Invalid chunked encoding: expected hex chunk size, "
+                    f"got {size_str!r} (upstream may have injected an error mid-stream)"
+                ) from None
             if chunk_size == 0:
```
[the remark this answers is not in the document]
Comment (Oaklight) [reply httpclient/httpclient.py:825]: Good call — fixed in d4793ce. Truncating to 100 bytes, matching what aiohttp does.

Here's a survey of how other libraries handle this:

| Library | Raw text in error? | Truncated? | Limit |
|---|---|---|---|
| **aiohttp** | Yes | **Yes** | **100 bytes** |
| urllib3 / h11 | Yes | No | (capped at read level: 64KB / 16KB) |
| Go net/http | No | N/A | 4096-byte line read cap |
| Node.js (llhttp) | No | N/A | Generic message only |
| Rust hyper | No | N/A | Generic message only |

RFC 9112 doesn't define a max chunk-size line length — 100 bytes is enough to show `HTTP/1.1 502 Bad Gateway` without dumping an HTML error page into logs.
